src/queries.js

```diff
@@ -10,156 +10,261 @@ function debugDOM(htmlElement) {
+    )}`,
+  )
+}
+
+function noMatchingLabel(container, text) {
+  return new Error(
+    `Unable to find a label with the text of: ${text} \n\n${debugDOM(
+      container,
+    )}`,
+  )
+}
+
 function getByLabelText(container, text, ...rest) {
   const el = queryByLabelText(container, text, ...rest)
   if (!el) {
     const label = queryLabelByText(container, text)
     if (label) {
-      throw new Error(
-        `Found a label with the text of: ${text}, however no form control was found associated to that label. Make sure you're using the "for" attribute or "aria-labelledby" attribute correctly. \n\n${debugDOM(
-          container,
-        )}`,
-      )
+      throw noMatchingFormControl(container, text)
     } else {
-      throw new Error(
-        `Unable to find a label with the text of: ${text} \n\n${debugDOM(
-          container,
-        )}`,
-      )
+      throw noMatchingLabel(container, text)
     }
   }
   return el
 }
 
+function getAllByLabelText(container, text, ...rest) {
+  const els = queryAllByLabelText(container, text, ...rest)
+  if (!els.length) {
+    const labels = queryAllLabelByText(container, text)
+    if (labels.length) {
+      throw noMatchingFormControl(container, text)
+    } else {
+      throw noMatchingLabel(container, text)
+    }
+  }
+  return els
+}
+
+function noMatchingText(container, text) {
+  return new Error(
+    `Unable to find an element with the text: ${text}. This could be because the text is broken up by multiple elements. In this case, you can provide a function for your text matcher to make your matcher more flexible. \n\n${debugDOM(
+      container,
+    )}`,
+  )
+}
+
 function getByText(container, text, ...rest) {
   const el = queryByText(container, text, ...rest)
   if (!el) {
-    throw new Error(
-      `Unable to find an element with the text: ${text}. This could be because the text is broken up by multiple elements. In this case, you can provide a function for your text matcher to make your matcher more flexible. \n\n${debugDOM(
-        container,
-      )}`,
-    )
+    throw noMatchingText(container, text)
   }
   return el
 }
 
-function queryByAltText(container, alt) {
-  return (
-    Array.from(container.querySelectorAll('img,input,area')).find(node =>
-      matches(node.getAttribute('alt'), node, alt),
-    ) || null
+function getAllByText(container, text, ...rest) {
+  const els = queryAllByText(container, text, ...rest)
+  if (!els.length) {
+    throw noMatchingText(container, text)
+  }
+  return els
+}
+
+function noMatchingAltText(container, alt) {
+  return new Error(
+    `Unable to find an element with the alt text: ${alt} \n\n${debugDOM(
+      container,
+    )}`,
   )
 }
 
 function getByAltText(container, alt) {
   const el = queryByAltText(container, alt)
   if (!el) {
-    throw new Error(
-      `Unable to find an element with the alt text: ${alt} \n\n${debugDOM(
-        container,
-      )}`,
-    )
+    throw noMatchingAltText(container, alt)
```

Comment:
What if instead of creating these error functions we had the `getBy*` functions use the `getAllBy*` functions? Perhaps using `firstResultOrNull`.

Sorry, I still haven't reviewed all the PR but this is just something else I noticed.

Reply:
Yep that makes sense!